libraries/AP_GPS/AP_GPS_GSOF.h

```diff
@@ -48,10 +48,10 @@ class AP_GPS_GSOF : public AP_GPS_Backend
     uint32_t SwapUint32(uint8_t* src, uint32_t pos);
     uint16_t SwapUint16(uint8_t* src, uint32_t pos);
 
-
     struct gsof_msg_parser_t
     {
-        enum
+
+        enum class gsof_states
```

Comment:
Isn't `gsof_` redundant here?

Also, I prefer the singular rather than plural for the enum class.

Reply:
Done and done. f29b433e8e548875beecc787b30fa032fa909323

Before: 
`gsof_msg.gsof_state = gsof_msg_parser_t::gsof_states::STARTTX;`

After:
`gsof_msg.state = msg_parser_t::state::STARTTX;